cmd/notary/keys.go

```diff
@@ -449,12 +430,11 @@ func (k *keyCommander) keysRotate(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	for _, role := range rolesToRotate {
-		if err := nRepo.RotateKey(role, k.rotateKeyServerManaged); err != nil {
-			return err
-		}
+	err = nRepo.RotateKey(rotateKeyRole, k.rotateKeyServerManaged)
+	if err == nil && k.rotateKeyServerManaged {
+		err = nRepo.Publish()
```

Comment:
We'd need to break up `Publish` into a few component steps but that wouldn't be a bad thing.